src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/CSharpCodeWriter.cs

```diff
@@ -26,44 +26,155 @@ public class CSharpCodeWriter : CodeWriter
             '\u2029',
         };
 
-        public new CSharpCodeWriter Write(string data)
-        {
-            return (CSharpCodeWriter)base.Write(data);
-        }
+        private static readonly char[] NewLineCharacters = { '\r', '\n' };
 
-        public new CSharpCodeWriter Indent(int size)
-        {
-            return (CSharpCodeWriter)base.Indent(size);
-        }
+        private string _cache = string.Empty;
+        private bool _dirty;
+
+        private int _absoluteIndex;
+        private int _currentLineIndex;
+        private int _currentLineCharacterIndex;
+
+        public StringBuilder Builder { get; } = new StringBuilder();
```

Comment:
Hmm if they can't see the builder, how will they know what got written? Eg: When I am debugging I do `Builder.ToString()` to check what got written. Is this something we care about?